src/parallel/boundaries/BoundaryHandler.cpp

```diff
@@ -0,0 +1,259 @@
+/*
+ * BoundaryHandler.cpp
+ *
+ *  Created on: 24 March 2023
+ *      Author: amartyads
+ */
+
+#include <algorithm>
+
+#include "BoundaryHandler.h"
+
+#include "Simulation.h"
+#include "integrators/Integrator.h"
+
+#include "utils/Logger.h"
+
+BoundaryHandler::BoundaryHandler()
+    : _boundaries{{BoundaryUtils::DimensionType::POSX,
+                   BoundaryUtils::BoundaryType::PERIODIC},
+                  {BoundaryUtils::DimensionType::POSY,
+                   BoundaryUtils::BoundaryType::PERIODIC},
+                  {BoundaryUtils::DimensionType::POSZ,
+                   BoundaryUtils::BoundaryType::PERIODIC},
+                  {BoundaryUtils::DimensionType::NEGX,
+                   BoundaryUtils::BoundaryType::PERIODIC},
+                  {BoundaryUtils::DimensionType::NEGY,
+                   BoundaryUtils::BoundaryType::PERIODIC},
+                  {BoundaryUtils::DimensionType::NEGZ,
+                   BoundaryUtils::BoundaryType::PERIODIC},
+                  {BoundaryUtils::DimensionType::ERROR,
+                   BoundaryUtils::BoundaryType::ERROR}} {}
```

Comment:
I'll change the init location

The last entry is only for sanity, if any function returns DimensionType::ERROR and it is looked up in the lookup table without further checks, we would get a BoundaryTyoe::ERROR. But this can be handled elsewhere more gracefully, so I'll do that.